pyiceberg/table/__init__.py

```diff
@@ -1290,6 +1291,17 @@ def snapshot_by_name(self, name: str) -> Optional[Snapshot]:
             return self.snapshot_by_id(ref.snapshot_id)
         return None
 
+    def snapshot_at_or_before_timestamp(self, timestamp_ms: int) -> Optional[Snapshot]:
+        """Get the snapshot that was current at or right before the given timestamp, or None if there is no matching snapshot."""
+        for log_entry in reversed(self.history()):
+            if log_entry.timestamp_ms <= timestamp_ms:
+                return self.snapshot_by_id(log_entry.snapshot_id)
+        return None
+
+    def current_ancestors(self) -> Iterable[Snapshot]:
+        """Get a list of ancestors of and including the current snapshot."""
+        return ancestors_of(self.current_snapshot(), self.metadata)  # type: ignore
```

Comment:
```suggestion
        return ancestors_of(self.current_snapshot(), self.metadata)
```
I think we do not need this any more?

Reply:
I've resolved the ignore error.

I think we don't need `current_ancestors()` anymore. I'm not using it in this PR, nor do we need this as a public method, since table.history() does a very similar job. 

The `ancestors_of()` is no longer being used, but as mentioned in previous discussions in this PR, it is a small function that is used in the cherrypick PR, once these changes are merged.

Reply:
+1 on removing `current_ancestors()` if we do not need it. I think we can keep `ancestor_of()` since it will be an useful helper. We just need to make sure there are tests for it. (Which you have already added for `current_ancestors()`:raised_hands:).

Reply:
Thanks for working on this @chinmay-bhat, I also [ran into the confusion around `current-ancestor` myself](https://github.com/apache/iceberg/issues/6930). Thanks for adding that to the docs itself.